src/Compilers/Core/Portable/CommandLine/SarifV2ErrorLogger.cs

```diff
@@ -68,17 +68,14 @@ public override void LogDiagnostic(Diagnostic diagnostic)
                 _writer.WriteObjectStart(); // suppression
                 _writer.Write("kind", "inSource");
 
-                if (Compilation != null)
+                object? justification = diagnostic.SuppressionInfo?.Attribute
+                                                                    ?.NamedArguments
+                                                                    .FirstOrDefault(o => o.Key == "Justification")
```

Comment:
Are keys meant to be case sensitive?

Reply:
Yes, they are. Updated the arguments retrieve way, the API of `DecodeNamedArgument` says it's case sensitive and using 'justification' would cause a compiler error was validated.